### PR TITLE
Fix API for new Intra-Process Communication

### PR DESCRIPTION
**Pull request by alsora**
Some small fixes to be compatible with new Intra-Process Communication 

Connects to ros2/rclcpp#778

@ivanpauno

### CODE REVIEW

File: rosbag2_transport/src/rosbag2_transport/generic_publisher.hpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ class GenericPublisher : public rclcpp::PublisherBase
     const std::string & topic,
     const rosidl_message_type_support_t & type_support);
 
-  ~GenericPublisher() override = default;
```
[the remark this answers is not in the document]
Comment (alsora) [reply rosbag2_transport/src/rosbag2_transport/generic_publisher.hpp]: Now `PublisherBase` class has not anymore a virtual destructor

**Comment by alsora** [comment]
Yes exactly.
This PR will be required because some API in `PublisherBase` and `SubscriptionBase` have been changed/removed

File: rosbag2_transport/src/rosbag2_transport/generic_publisher.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Karsten1987) [reply rosbag2_transport/src/rosbag2_transport/generic_publisher.hpp]: then the GenericPublisher should either define its own `virtual ~GenericPublisher() = default` or being marked as `final`.

[the remark this answers is not in the document]
Comment (alsora) [reply rosbag2_transport/src/rosbag2_transport/generic_publisher.hpp]: Updated

[the remark this answers is not in the document]
Comment (wjwwood) [reply rosbag2_transport/src/rosbag2_transport/generic_publisher.hpp]: I think this needs to be updated again, see: https://github.com/ros2/rclcpp/pull/778#discussion_r326430357

[the remark this answers is not in the document]
Comment (wjwwood) [reply rosbag2_transport/src/rosbag2_transport/generic_publisher.hpp]: Well, I guess leaving it as-is with `virtual` is fine actually.

File: rosbag2_transport/src/rosbag2_transport/rosbag2_node.cpp [hunk 2]
```diff
@@ -56,7 +56,9 @@ std::shared_ptr<GenericSubscription> Rosbag2Node::create_generic_subscription(
       topic,
       callback);
 
-    get_node_topics_interface()->add_subscription(subscription, nullptr);
+    // Intra-process message handling has to be disabled.
+    bool use_intra_process = false;
+    get_node_topics_interface()->add_subscription(subscription, nullptr, use_intra_process);
```
Comment (wjwwood) [inline rosbag2_transport/src/rosbag2_transport/rosbag2_node.cpp]: I think this needs to be updated now, I removed the `use_intra_process` argument from this function.

Comment (wjwwood) [reply rosbag2_transport/src/rosbag2_transport/rosbag2_node.cpp]: I fixed this and rebased.

**Pull request merged**